control-plane/subcommand/server-acl-init/command.go

```diff
@@ -592,7 +592,15 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 		serviceAccountName := c.withPrefix("api-gateway-controller")
-		if err := c.createACLPolicyRoleAndBindingRule("api-gateway-controller", rules, consulDC, primaryDC, localPolicy, primary, localComponentAuthMethodName, serviceAccountName, consulClient); err != nil {
+
+		// API gateways require a global policy/token because they must
+		// discover services in other datacenters.
```

Comment:
```suggestion
		// create config-entry resources in the primary, even when deployed to a secondary datacenter
```
Isn't ^ the actual reason why this is necessary? We aren't enabling routing to services in other datacenters yet.

Reply:
You're right, good catch!